### ISSUE TITLE
Release 2.19.0 makes database passwords being interpreted differently than with MySQL leading to authentication errors

### ISSUE THREAD

**Issue by AstroGD** [kimai/kimai#4997]
### Describe the issue

The pull request https://github.com/kimai/kimai/pull/4976 introduced changes to Kimai allowing special characters in the database password. However, it introduced the side effect of not allowing the `+` symbol to be used in the password, as it will be interpreted as a whitespace character. This is unexpected behavior and works contrary to, for example, how the password string would be interpreted by MYSQL used with Kimai.

Example: The password `abc123+abc123` would be interpreted as `abc123+abc123` within MySQL but as `abc123 abc123` with Kimai leading to an authentication error when using the same environment variable for MySQL and Kimai.

I don't know why this change was needed, but I highly suggest reworking it so that entering the same password into MySQL and Kimai doesn't lead to it being interpreted differently.

### I already tried

- [X] I've read and searched [the documentation](https://www.kimai.org/documentation/).
- [X] I've searched for similar issues in this repository.
- [X] I've searched for similar issues in [the discussions](https://github.com/kimai/kimai/discussions).

### Kimai version

2.19.0

### How do you run Kimai?

Docker

### Which PHP version are you using?

Unknown

### Logfile

```
Testing DB:** new \PDO(mysql:host=db;dbname=database;port=3306, username, abc123 abc123, [ \PDO::ATTR_ERRMODE => \PDO::ERRMODE_EXCEPTION ]);*Access denied (1045)Checking DB: 1
+ php /dbtest.php db dbname 3306 username abc123+abc123
+ echo Checking DB: 1
+ sleep 3
```

**Comment by kevinpapst** [comment]
Oh snap, I think you are right. It should have been urlencode, not urldecode...

Also it is a breaking change for existing users. Will revert.

**Comment by kevinpapst** [comment]
Released 2.19.1. can you test?

**Comment by kevinpapst** [comment]
Next release will bring that back. You have to encode your database URL as documented.
